### ISSUE TITLE
Meta: Self-Assembling Organization

### ISSUE THREAD

**Issue by cjslep** [git-federation/gitpub#5]
This is going to get a bit meta, but it is just because I am biased to wanting to see success here.

The first part A is kind of an overview of the different moving pieces I see going on, which others may or may not already be aware of. My intention here is to be informative about the state of this space, not dictatorial/authoritative. Part B is just my bland appeal to interested parties in making this successful in a coordinated fashion.

**Part A**

Just from the various topics I've seen float around, it seems there's several problems to solve, different prioritization in which to solve these problems, different ideas on what those solutions precisely look like, and (obviously) different parties doing these.

**What**
- *Problem of federated authorization*: This was talked about in the #social irc some as mentioned in #2
- *Problem of fitting Git socialization data into the ActivityStreams data model*: Mentioned in #1. Maybe a part of #2, also?
- *Problem of fitting Git socialization behaviors into the ActivityPub actor model*: Also mentioned in #1. Maybe a part of #2, also?

**When**
- I've seen suggestions elsewhere of rolling out in two phases: First provide data as ActivityStream objects, then work on the consumption side.
- I've seen some prioritize for solving the federated authorization first.
- The Git socialization features would realistically be implemented piece by piece as well, but no timelines until the design is done.

**How**
- For federated auth, I've seen IndieAuth and ocap-ld suggested.
- For both Git socialization problems, there's things documented in #1 and #2 already.

**Who**
- SocialWG: Authored the [ActivityStream 2.0 Spec](https://www.w3.org/TR/activitystreams-core), [ActivityStream 2.0 Vocab](https://www.w3.org/TR/activitystreams-vocabulary), also maintains the [ActivityStream 2.0 terms list](https://www.w3.org/ns/activitystreams), also authored the [ActivityPub spec](https://www.w3.org/TR/activitypub). As well as many others.
- All the git hosting solution providers: Gitea, Gogs, Gitlab, etc
- Other ActivityPub implementors, such as Mastodon, etc
- (Sorry if I missed you!)
- Peanut gallery w/ biased, vested interest (I fall here).

**Why**
I had a recent back and forth on Mastodon with a skeptic about this effort, so forgive me for not including this at this time. :)

Hopefully the above is informative!

**Part B**

I'd love for this effort across projects & people to succeed. I would hate to see people left out of conversations, problems not getting enough visibility, or piecemeal solutions adopted.

I don't have a good solution here; I'm new to the self-organizing open source space. Since I am not a part of the SocialWG nor any of the user-facing projects, I also don't know what the right self-organizing communication structure would look like to get buy-in on decisions from everyone.

Part of this appeal stems from seeing issues like #4, part of it is fueled by my raw excitement. Thanks for reading this long post.

**Comment by neithernut** [comment]
Although you call this "Meta", it does contain traces about what should actually go into a design document:
 * A list of share-holders
 * A formalization of the requirements

These would be starting points, rather than the underlying technologies (e.g. ActivityStream) in classical development. I feel they should be clarified rather sooner than later. One of the maintainers could open a PR which would be used for assembling and discussing these basic points. It'll be probably be a mess, but better have the mess now than later, when different expectations and ideas clash in the specification of the protocol or, worse, the implementation.

**Comment by yookoala** [comment]
A quick note on point 1 in "What": SNS like Mastodon federates their identity by including the server domain to it (i.e. "yookoala@github.com" for my account here). The same username in different server is considered different users and thus no need to reserve username across federation.

**Comment by techknowlogick** [comment]
Personally I am against google groups, but not so stubborn, as I would use it if a majority agrees.

I've seen loomio (as mentioned above) used successfully, and would vote for that. Although, as @bill-auger linked to they were able to have these discussions in a git repo, and even Rust-lang uses git repos for their RFCs (example: https://github.com/rust-lang/rfcs/pull/2457), so I also wouldn't be opposed to using this repo for these discussions (all potential members have GitHub accounts currently).

**Comment by bill-auger** [comment]
google does not have mailing lists - google has "google groups" web forums - i quite literally meant a "mailing list" (as in: i can read and receive messages in my email) (and without clicking my mouse)

**Comment by bill-auger** [comment]
> "(all potential members have GitHub accounts"

i would not assume so much - "all potential members" is the entire world isnt it?
all that is evident this far is that all people who have expressed an interest in this group have github accounts - surely, that is no co-incidence as this group has offered no other contact information and is so far limiting all discussions to github

**Comment by techknowlogick** [comment]
@bill-auger I apologize, the "all potential members" comment was in reference to the list above titled `Potential Members in Work Group`, not in reference to everyone in the world.

**Comment by bill-auger** [comment]
yes i assumed that - but really this group was just formed about 12 hours ago so sure there are more potential members - this was a hot topic of discussion on freenode yesterday - many people are interested in this topic and FWIW many closed their gthub accounts yesterday BTW

**Comment by yookoala** [comment]
@bill-auger: Are there any good (and free :-) ) mailing list service out there we might use?

**Comment by bill-auger** [comment]
off-hand i know of tuxfamily and savannah - possibly riseup or frammasoft

**Comment by yookoala** [comment]
I heard good comments on framasoft's service ([Framalistes](https://framalistes.org)). If there is no objections, I'd start a list there tomorrow :-)

**Comment by 21stio** [comment]
Maybe let's do a voting, one comment per option and everybody thumbs up their preference. Maybe options should be formatted in a standardized way.

Would be nice if each option would provide a one sentence description and how much it costs, as loomio is only free for up to 10 people (wouldn't mind if somebody creates a gitsub patreon to finance it tho)

**Comment by simon-brooke** [comment]
Seriously, folks, letting centralised capitalist entities host our repositories is how we got into this mess in the first place. We need a decentralised, mastodon.social style, federation of nodes, all or most of which are controlled by individual developers.

The bits of this are simple. We need 

1. a githook post-commit script that publishes an ActivityPub message with a specified format on each commit;
2. an ActivityPub consumer which reads commit messages from chosen streams and automatically does a pull request - so as to provide backup if the 'home node' of a project goes dark;
3. a search spider which actively seeks out nodes and indexes the projects they host;
4. a search interface which allows users to query what has been discovered by the spider.

It's all simple, we can do this.

**Comment by 21stio** [comment]
@simon-brooke how is your comment related to the organisational setup? please open an issue if want to talk about implementation details imho

**Comment by bill-auger** [comment]
21stio - how could it possibly be preferable to pay a centralized host and require everyone to sign-up for an account on it (presumably most people have never even heard of that site) merely for the sake of something as simple a discussion - to me that is less sensible than simply continuing to use this issue tracker - rather than using email which is is federated, costs nothing, and everyone has it setup already - and if for some *very* peculiar reason, someone does not have email yet (an email address is of course a requisite for both git and github) that person is free to choose their email provider or self-host it

dont you see the folly in that? - it is ironic beyond belief that such suggestions of relying on yet another third-party proprietary service would be offered in this context - the only reason i am in this conversation is to introduce this group to some people who are very passionate about federation and have been thinking about exactly this proposal of a federated git hosting platform for a long time and have already done most if the work - but such people who are so passionate about federation simply refuse to use *any* third-party proprietary services; so suggesting that this group should use one for communication is essentially saying "im do not want the people who are most passionate about federation to be in this conversation"

**Comment by bill-auger** [comment]
simon-brooke's first paragraph is precicely on topic - i was elaborating on that same point myself at the time it went up

simon-brooke issue #1 is a discussion about spec proposals - i added there the notabug-2.0/vervis design document - you may be interested to see that most of the work has been done to this end and the vervis reference implementation is nearly ready for demo https://notabug.org/NotABug.org/notabug-2.0

this issue is only about where to find a home for the project - i have offered tuxfamily as one option because in addition to mailing lists, they offer full hosting in case folks like some mattermost sort of things eventually - at this point, i still content that email is more than sufficient at this early stage

**Comment by yookoala** [comment]
@bill-auger: I think @21stio meant to pointe out that this thread is not about specification or implementation details. Although @simon-brooke has valuable input, it is not about forming a work group, which is the title of this issue.

@simon-brooke: Couldn't agree more that we want to prevent relying on centralized hosted service. I've proposed to extend ActivityPub precisely because of that. I've specify how it could work in the [current draft branch](https://github.com/git-federation/gitpub/blob/draft-0.1/SPECIFICATION.md). But in the end, its more important to have enough stack-holders agree on an open standard than my opinion. Feel free to start a new issue to discuss your proposal. And feel free to join the work group mailing list (which would be created tomorrow).

**Comment by 21stio** [comment]
@bill-auger

I never stated any preferences. There are mixed opinions on where this should live. Some prefer an ideological biased solution, some a pragmatic. I'd lean towards a pragmatic one but I don't really have an opinion, Im fine with the choice of the majority. It would simply be nice if we would find a consent ASAP.

**Comment by pypingou** [comment]
:+1: to framalist, classic mailing list remain the most accessible mean of communication and discussion in a group.

**Comment by bill-auger** [comment]
isnt a decentralized git hosting network an ideological solution? and arent centralized git "hubs" the "pragmatic" solution? so why exactly are we here then?

i would not put it in terms of ideology vs pragmatism - its simple dogfooding and it gives outsider confidence in the organization that they truly believe what they say and they practice what they preach

**Comment by 21stio** [comment]
to me pragmatic simply means getting stuff done as efficient as possible, by using stuff everybody is familiar with

and I'd only lean towards a pragmatic solution in the context of project management/communictation at this point in time

I didn't state anything about any other context and any other point in time, with my initial comment I only made a suggestion on how we can have a decision, I'd be very happy if we could end this conversation please

**Comment by jaywink** [comment]
Personally I would prefer keeping discussion here in one place, until we can move this repo to something that federates. But failing, that, if another communication medium is needed, 👍 to an old traditional mailing list, if such is available for example from framalist. This would be the most neutral and allow for the easiest participation.

Also, I would in theory very much like to participate in the working group, my experience is:
* Previously Diaspora core team member (~2014-2016)
* W3C SocialWG workgroup (that produced ActivityPub) member as invited expert
* Socialhome federated social network author (so I have written a library for Diaspora protocol, planning ActivityPub support)

I'm super excited by this and having federated code hosting has been my dream for years.

**Comment by 21stio** [comment]
I'd also like to participate in the Tentative Work Group

**Comment by pypingou** [comment]
FTR, I have no opinion b/w framalist or tuxfamily, mailing lists are mailing lists, I don't really care where they are hosted :).

framavox is also a potential option, never used it so I can't provide feedback but I'd be willing to give it a try, I just don't know if it is something that can integrate well in a email-based workflow or if I'll need to keep a tab open with it.

**Comment by coderkevin** [comment]
Hi there, I just came across this from the gitlab federated discussion. I'm interested in being involved here as I could see this being the future home of OSS projects. I don't have experience with SocialWG or ActivityPub, or even writing specifications, I've just been writing code for a couple decades and can help with that. Is there room for me here?

For the discussions/proposals format, I'd think prioritizing using one's own platform would be key. So after this federated git thing is built, using that. And before it's ready, anything that ports/exports easily to it would work (so anything using git + branch review like pull/merge requests). So, using something like GitHub or GitLab until this thing is ready, then importing the entire git history and using the new thing. It'd be nice to have the full history intact.

**Comment by cjslep** [comment]
I'm abstaining from providing input on the choice of tools -- I defer to those more experienced & opinionated than I.

I do want to emphasize @bill-auger's point about people being very enthusiastic about solving problems already. It's pretty impressive, and I hope spirits stay high. However, these quick-moving discussions also show the need for a decision-making process. It looks like Committee Voting? (We sure we don't want a BDFL? Hehe, j/k) Who are the *stakeholders* that get a vote? A vote per person, or per project?

Then the questions about thresholding. First Past The Post (50%, 2/3)? Single Transferrable Vote? Ranked ballots?

Or is this too formal, and we just play it by general consensus?

**Comment by 21stio** [comment]
What about simply staying here for now. A lot of people already starred and are watching the repo, link juice is building up, everybody knows the tooling and has an account.. 

What feature are we missing here (besides federation)?

**Comment by jaywink** [comment]
+1 for staying here. And in regards any possible voting (bound to happen :)), I would favour for example voting with +1/-1's in the issue as reactions to a proposal, or some other way in writing. Voting in meetings excludes those who can't participate at that particular time.

**Comment by yookoala** [comment]
@cjslep: I think it makes sense to have one vote per project. But in reality, I don't think there's a reasonable way to decide who get to vote in a large open source project. So that would be hard to implement.

I personally prefer to have general consensus as the default resolution method. But I do think we need a way to vote when undecided. So I'd prefer to have a tool to cater for such situations. And in such situations, I think it makes more sense for people to vote only if they have participated in the discussion. So a simply "+1" and "-1" in Github comment might be too chaotic for management.

So I'd rather we have a voting platform setup for people who joined our mailing list. And probably a simple majority (50%) vote? What do you all think?

**Comment by yookoala** [comment]
### Mailing List

I have created a mailing list on Framalistes. And my email is *koalay at gmail.com*. I have closed down public subscription there. I hope people who join the group would at least introduced himself here.

I think I managed to find email address for @cjslep and @techknowlogick but not everyone else. Please send me a email and state your Github user name in it. I'll add you all to the list.

### Tentative Work Group Members
Someone I think have shown clear interest here in this thread.
Please correct me if I'm wrong.

(alphabetical order)
@21stio 
@bill-auger
@cjslep
@coderkevin 
@cwebber
@jaywink
@pypingou
@samheutmaker 
@techknowlogick
@yookoala

Welcome to join later anyway.

### Decision Making

Since people are very enthusiastic towards the federation idea, I think we need to setup some voting mechanism quickly before all the inputs get out of hand. I'm proposing to use [Framavox](http://framavox.org/) for voting in the future. I think this should be our first general consensus discussion in the mailing list.

**Comment by Unknwon** [comment]
+1 for staying here.  And I personally suggest 66.67% vote. 50% agree could mean argument points are even on both sides (even could potentially have three and more sides on one idea).

**Comment by bill-auger** [comment]
this repo does not need to go away - no one is suggesting that - you will probably need a place to hold the documents - the mailing list is for discussions - that is what mailing lists are for - this is a bug tracker - it could surely be left open for that purpose

the mailing list has been created and the majority have already agreed that is the most universal communication tool that will invite the most participants - if the discussion stays on github you are guaranteeing that only github fans will be in the discussions; which is extremely mis-aligned with the goals of this group

**Comment by yookoala** [comment]
@Unknwon: Thanks. If found your email on your Github profile and I've just added it to the mailing list.

**Comment by theodotos** [comment]
I'm also interested in the working group. I'm a lousy coder but can help with sysadmining and testing. Can I be added in the mailing list too?

**Comment by yookoala** [comment]
@theodotos: Sure. I might be an even lousier coder :-)

Just send me an email so I know your email address. My email address is marked on [this comment](https://github.com/git-federation/gitpub/issues/5#issuecomment-395283212).

**Comment by Arkanosis** [comment]
@yookoala could you please add me as well to the ML (jroquet@arkanosis.net)? Thanks!

**Comment by yookoala** [comment]
@Arkanosis: Just added. Please check your mailbox to see if you received the invitation from Framalistes.

**Comment by yookoala** [comment]
@Unknwon: The Framalistes backend said that your email at gogs.io is bouncing. Please check if you got the invitation. If not you'd probably need to give me another email address. Just send me a mail to *koalay at gmail.com* to discuss. Thanks.

**Comment by Foxboron** [comment]
Yo.

Interested discussing and helping the effort. Experience with semantic technologies and vocabularies from University, along with a lot of git experience and avid programmer. Would be awesome to be added to the mailing list for further discussions. morten@linderud.pw

**Comment by yookoala** [comment]
@Foxboron: Just added. Please check your mailbox for invitation. Thanks.

**Comment by Foxboron** [comment]
Got it. Thank you :)

**Comment by b12f** [comment]
Can I be added to the ML? Also interested in helping / discussing wherever possible.  hello@benjaminbaedorf.eu

**Comment by yookoala** [comment]
## Notice about Joining the Mailing List

All: I feel that the mailing list is about the right size to have diversify opinions without making everyone crazy. I think I'm going to set some boundaries for later joiner:

If you are a **git service software developer**, please introduce yourself here and state that you want to join. If **nobody in the current mailing list disagree**, we'd add you to the mailing list.

This is only for the sake of discussion quality. If any mailing list member disagree, we may discuss over the mailing list. And if you're not a member, you're welcome to file a new issue in this issue tracker for discussion.

Thanks a lot.

**Comment by yookoala** [comment]
@Unknwon: I've checked the bouncing message. Our mailing list server got "550 Mail content denied" from QQ's mail server. It could be just spam filter or it could be something else entirely. Do you have any extra mailbox (e.g. say, Gmail?) that we can contact you with? Thanks.

**Comment by goffi-contrib** [comment]
Hi there, this repository has been pointed to me on our chat room.

For your information, I've worked on decentralized tickets and merge-requests system based on XMPP and it's already working (cf. https://www.goffi.org/b/9555cc02-6a87-4b6b-af85-20f1c0736722 and https://www.goffi.org/b/F4xScokjZejCYAB4NamBbc/decentralized-code-forge,-based-xmpp). In other words, there is working basis for decentralized code forge.

It's not restricted to Git (we are using it with Mercurial), and it's possible to do gateway to existing plate-form.

You can ping me if you wants more details.

**Comment by lafriks** [comment]
Please add me also to mailing list

**Comment by bill-auger** [comment]
lafriks -

firstly, you did not give your email address - so adding you to the list would be quite impossible

but more importantly, yookoala has stated that the list should remain as small as possible but large enough only to have enough diverse opinions and backgrounds - you would really need to give some description of your qualifications and specific interest in the group

**Comment by neithernut** [comment]
His mail address is stated in his profile. And it appears that he's a Gitea contributor. It would still have been nice if this was stated by himself rather than requiring some investigation.

**Comment by lafriks** [comment]
Sorry for not giving more detailed info, I was on mobile phone :) So as said in previous comment. yes my email address is what is in my GitHub account. I'm Gitea contributor and currently one of owners.

**Comment by yookoala** [comment]
@lafriks: Just add your another email to the mailing list. Please check your mailbox and see if you get the invitation. Thanks.

**Comment by Unknwon** [comment]
@yookoala I think I have received emails, I've added sender to whitelist.

**Comment by yookoala** [comment]
> I think I have received emails, I've added sender to whitelist.

@Unknwon: Nice! thanks.

**Comment by arucard21** [comment]
> If you are a git service software developer, please introduce yourself here and state that you want to join.

I just want to raise a concern regarding this. You may want to include more people, like software developers with experience using git collaboratively. I'm sure that the developers working on a git service also use that service for their development, but their usage might differ from regular users or might only cover a small subset of the use cases. 

Of course, to keep quality of the discussion high you should still try to ensure that they have sufficient knowledge about git, git services, federation techniques, usage and modification of standards and protocols, etc. Some of this knowledge doesn't even require a developer to be familiar with git or git services at all. For example, I'm sure you'd want someone who helped create ActivityPub to be included in the discussion, even if that person has never used git or any git service before. 

I hope that this diversity of knowledge and experience can help create better git federation. Best of luck to you all.

Disclaimer: I'm not a git service software developer but I also don't need to join the mailing list. So I think I can be fairly objective about this.

**Comment by bill-auger** [comment]
there is a discussion of those concerns on the mailing list now

i will add that the constraint you mention was added only after the group was already at significant size

**Comment by jaywink** [comment]
I think it's sad if the group decides to work in private and ignore public feedback. I would encourage, even if it adds noise, to either allow discussion here (by not actively discouraging it) or allow anyone to join the mailing list. For example the SocialWG had two mailing lists, one for public comments, one for group members. There has to be a way for the "public" to give feedback, otherwise it's not an open specification worked on openly, and that risks ignoring valuable feedback from someone who had some.

I realize noise is a problem but hiding from it is the wrong answer :)

**Comment by bill-auger** [comment]
that is the very reason this github issue tracker was left open for comments - it says so on the README now

**Comment by bill-auger** [comment]
@jcgruenhage -

the mailing list is world readable - the URL is on the README

**Comment by Foxboron** [comment]
@bill-auger It is possible to have a whitelisted mailinglist. So everyone can subscribe and only whitelisted people can post towards it. I believe that is what @jcgruenhage wants? Unsure if our current provider allows that.

**Comment by jcgruenhage** [comment]
@bill-auger I'm sorry, I searched framalists.org for gitpub and it said no results, so I assumed it was private. Thanks for pointing me in the right direction 👍 

@Foxboron yep, that would be exactly what I want

**Comment by yookoala** [comment]
> I just want to raise a concern regarding this. You may want to include more people, like software developers with experience using git collaboratively. I'm sure that the developers working on a git service also use that service for their development, but their usage might differ from regular users or might only cover a small subset of the use cases.

@arucard21: I think limiting the members of the mailing list is not the same as closing down the whole discussion from people other than Git service developers:
1. Some people in the mailing list is merely Git users. There are also people from previous ActivityPub specification process.
2. Our issue tracker here is always open to all. So anybody can brainstorm anything, or voice out at any stage of the specification process. Me and other mailing list members are still reading here. And we'll take anything new and insightful here to the discussion.
3. Our [discussion archive](https://framalistes.org/sympa/arc/git-federation/) is open to public. So anybody can read about our discussion and ask questions here.

My main concern, as stated, is the discussion quality. Not hiding from feedbacks.

**Comment by Foxboron** [comment]
@yookoala I sadly have no experience with actually setting one up :/

**Comment by arucard21** [comment]
@yookoala With the mailing list just being part of the discussion, this seems fair enough. And they are readable by anyone. Perhaps it might help the public part of the discussion to periodically provide updates in the relevant issues in the tracker. Or if something close to a consensus has been reached on the mailing list, maybe then open it up to the public by updating the relevant issue. 

But I'm glad to hear that in general you're trying to remain quite open. It's always hard to have these discussions openly and still have them be productive. Perhaps there are some open-source communities that have experience with this. They might have some ideas on at least avoiding some common mistakes.

**Comment by yookoala** [comment]
@arucard21: I have updated the notice to add link to the mailing list archive. I've also explained how people should use this issue tracker for feedback. I hope that resolved the concerns here.

**Comment by Foxboron** [comment]
We could write up summaries of discussions, issues and current proposals and stuff them into a git repository. This could either be done weekly, or monthly depending on the activity we end up with.

I know [reproducible builds](https://reproducible-builds.org/blog/) does this and it works quite well.

**Comment by yookoala** [comment]
@Foxboron: That's a nice suggestion. I think I'll try to do a weekly summary. So the first one should probably come on next Tuesday (?).

**Comment by arucard21** [comment]
@yookoala 
> I hope that resolved the concerns here.

It does (to me at least). Thanks.

**Comment by ckuttruff** [comment]
Hey all, read through comments carefully and meant to subscribe for updates only, not to actively post.  I am very interested in this project.  No direct work on git-specific services but background in general programming / devops.  Would love to follow discussion and contribute if there's a good way for me to do so.

But yeah, just want to subscribe to stay up to date.  Apologies if I didn't do that properly.  :+1: to discussion around public visibility.  I see the list archive is public, but it's much harder to set reminders to check back, so a read-only option for list subscription sounds perfect (with access poss by reaching out to existing members or via gh issues)

Thanks all for this interest and movement.  Would love to see this project come to fruition.  Obviously necessary

Btw I'm: chris *At* orghub Dot _co_

**Comment by jaywink** [comment]
@yookoala resolved for me too, sorry for the later reply, was sick for a few days :v:

**Comment by LWFlouisa** [comment]
I'm curious by what you mean by git developer. With the recent news about Github, I make my own personal hub in Beaker Browser to host my own gits.

But not really sure if that counts. It's not a Github alternative in the traditional sense, as it's largely hard coded rather than "submitted". Or in the blogging world, what they call "Add New Entry."

**Comment by yookoala** [comment]
@ckuttruff: I tried to make a read-only mailing list for the workgroup mailing list but failed. Don't seem to find a proper solution. If I solve it, I'll add you there. Meanwhile, I'm writing a summary of our discussion so far. Please come back again. I'll post the link to a blog / medium account where you can follow.

**Comment by yookoala** [comment]
@LWFlouisa: I'm not sure. Seriously. I've read about Beaker Browser but am not familiar with it.

I think to federate your repositories with our proposing federation, you'd need to have:

1. A programable HTTP / HTTPS endpoint for you to expose proper feed (e.g. ActivityStream 2.0 JSON feed or XML) about your repository.
2. Your program to handle external push messages to you (e.g. POST request to your inbox).
3. Endpoint for outsiders to pull / push your repositories.

If Beaker Browser allow you to do all 3. And if you're planning to write a, preferable open sourced, implementation of (2), then I'll consider you as a git service developer.

**Comment by SirCmpwn** [comment]
Hi, I'm the maintainer of [sr.ht](https://meta.sr.ht), and I would like to subscribe to your mailing list.

**Comment by yookoala** [comment]
@SirCmpwn: I have just added <sir at cmpwn dot com> to the mailing list. Please check if you get the invitation. Thanks for joining :-)

**Comment by SirCmpwn** [comment]
Thanks! I'll be sending a long and controversial email shortly.

**Comment by SoniEx2** [comment]
I don't think ActivityPub is the best protocol for this and I don't believe in federating pull requests. I think we should do it more like this: https://cybre.space/@SoniEx2/101858151417423058

**Comment by SoniEx2** [comment]
like this: https://ganarchy.autistic.space/

**Comment by genofire** [comment]
PRs and issues are from one instance to another and not back ... so you arrows should not be bidirectional - that will solve it.

**Comment by SoniEx2** [comment]
those aren't PRs and issues - those are merges/pulls (as in `git merge`, `git pull`, `git fetch`, etc).

in fact I think having PRs is not a good idea at all anymore. they're inherently centralizing.

**Comment by SoniEx2** [comment]
also this: https://cybre.space/@SoniEx2/101972146630176235

**Comment by SoniEx2** [comment]
also this https://ganarchy.github.io/

**Comment by SoniEx2** [comment]
oh and this I guess https://cybre.space/tags/gan385e734a52e13949a7a5c71827f6de920dbfea43

**Comment by ShadowJonathan** [comment]
@SoniEx2 please dont pollute this conversation by posting multiple links twice, please put them in one message if possible.

> I don't believe in federating pull requests.
Please then don't comment on this project, you can safely ignore this project and disable the feature if or when it'd eventually land in gitea, don't sabotage it instead, please.

> (in [toot](https://cybre.space/@SoniEx2/101858151417423058))
> ...notice how there's a lot more cross-pollination...

Personally, I believe that's exactly the right thing, as collaboration bounces around more "evolutionary" between different servers, where different communities are contributing to a "strand" of a particular project, and pull requests therefore can be exactly that "cross-pollination" that betters and adds to each particular version of a project, fixes and features can be shared across all strands of such software.

**Comment by bill-auger** [comment]
Jonathan -

FWIW, there is no more conversation on this issue tracker - it
has not been used since about two years ago - if you are
interested in forge federation, join the forum:
https://talk.feneas.org/c/forgefed
